plugins/module_utils/network/sonic/config/mclag/mclag.py

```diff
@@ -180,19 +214,167 @@ def _state_deleted(self, want, have):
+                                delete_all_vlans[option] = True
+                                del_command[option] = have_cfg
+                            else:
+                                have_vlans = set(self.get_vlan_id_list(have_cfg['vlans']))
+                                want_vlans = set(self.get_vlan_id_list(want_cfg['vlans']))
+                                if have_vlans.intersection(want_vlans):
+                                    del_command[option] = {'vlans': self.get_vlan_range_list(list(have_vlans - want_vlans))}
+                                    if not del_command[option]['vlans']:
+                                        del_command.pop(option)
+                                    add_command[option] = {'vlans': self.get_vlan_range_list(list(want_vlans - have_vlans))}
+                                    if not add_command[option]['vlans']:
+                                        add_command.pop(option)
+                                else:
+                                    delete_all_vlans[option] = True
+                                    del_command[option] = have_cfg
+                                    add_command[option] = want_cfg
+                        else:
+                            if want_cfg['vlans']:
+                                add_command[option] = want_cfg
+                    else:
+                        if have_cfg and state == 'overridden':
+                            delete_all_vlans[option] = True
+                            del_command[option] = have_cfg
+
+                del_diff = get_diff(self.remove_default_entries(have), want, TEST_KEYS)
+                for option in del_diff:
+                    if not want.get(option):
+                        # Delete all non-specified configrations in overridden
+                        if state == 'overridden':
+                            del_command[option] = del_diff[option]
+                    else:
+                        # Delete portchannels that are not specified
+                        if option == 'members' and want.get(option):
+                            del_command[option] = del_diff[option]
+
+                        # To update 'gateway_mac' configuration in the device,
+                        # delete already configured value.
+                        if option == 'gateway_mac' and want.get(option):
+                            del_command[option] = del_diff[option]
+
+                diff = get_diff(want, have, TEST_KEYS)
+                add_command.update(diff)
+
+            if del_command:
+                del_command['domain_id'] = have['domain_id']
+                commands = [update_states(del_command, 'deleted')]
+                if delete_all:
+                    requests = self.get_delete_all_mclag_domain_requests(del_command)
+                else:
+                    if any(delete_all_vlans.values()):
+                        del_command = deepcopy(del_command)
```

Comment:
What is the reason for this instruction?

Reply:
It was added to avoid modifying 'have' from which 'del_command' is derived.